e2e/test/scenarios/admin/database-routing2.cy.spec.ts

```diff
@@ -0,0 +1,306 @@
+  before(() => {
+    cy.signInAsAdmin();
+    H.queryWritableDB(
+      "SELECT datname from pg_database WHERE datname = 'lead'",
+    ).then((res: { rows: any[] }) => {
+      if (res.rows.length === 0) {
+        H.queryWritableDB("CREATE DATABASE lead;");
+      }
+    });
+
+    H.queryWritableDB(
+      "SELECT datname from pg_database WHERE datname = 'postgres_two'",
+    ).then((res: { rows: any[] }) => {
+      if (res.rows.length === 0) {
+        H.queryWritableDB("CREATE DATABASE postgres_two;");
+      }
+    });
+
+    H.queryWritableDB(
+      "SELECT datname from pg_database WHERE datname = 'postgres_three'",
+    ).then((res) => {
+      if (res.rows.length === 0) {
+        H.queryWritableDB("CREATE DATABASE postgres_three;");
+      }
+    });
+
+    const leadConfig = {
+      client: "pg",
+      connection: {
+        ...QA_DB_CREDENTIALS,
+        port: QA_POSTGRES_PORT,
+        name: "lead",
+        database: "lead",
+      },
+    };
+
+    const aConfig = {
+      client: "pg",
+      connection: {
+        ...QA_DB_CREDENTIALS,
+        port: QA_POSTGRES_PORT,
+        name: "postgres_two",
+        database: "postgres_two",
+      },
+    };
+
+    const bConfig = {
+      client: "pg",
+      connection: {
+        ...QA_DB_CREDENTIALS,
+        port: QA_POSTGRES_PORT,
+        name: "postgres_three",
+        database: "postgres_three",
+      },
+    };
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: leadConfig,
+      query: "CREATE TABLE IF NOT EXISTS db_identifier (name VARCHAR(50));",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: leadConfig,
+      query: "DELETE FROM db_identifier;",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: leadConfig,
+      query: "INSERT INTO db_identifier VALUES ('lead');",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: aConfig,
+      query: "CREATE TABLE IF NOT EXISTS db_identifier (name VARCHAR(50));",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: aConfig,
+      query: "DELETE FROM db_identifier;",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: aConfig,
+      query: "INSERT INTO db_identifier VALUES ('postgres_two');",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: bConfig,
+      query: "CREATE TABLE IF NOT EXISTS db_identifier (name VARCHAR(50));",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: bConfig,
+      query: "DELETE FROM db_identifier;",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: bConfig,
+      query: "INSERT INTO db_identifier VALUES ('postgres_three');",
+    });
+
+    cy.task("connectAndQueryDB", {
+      connectionConfig: bConfig,
+      query: "SELECT name FROM db_identifier;",
+    });
+
+    H.setTokenFeatures("all");
+    // @ts-expect-error - this isn't typed yet
+    cy.createUserFromRawData(personA);
```

Comment:
These requests are failing due to auth, but I can't see why. I sign in as admin in the first line.